gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -857,6 +887,8 @@ public void onContainersAllocated(List<Container> containers) {
 
         ContainerInfo containerInfo = new ContainerInfo(container, instanceName, containerHelixTag);
         containerMap.put(container.getId(), containerInfo);
+        allocatedContainerCountMap.putIfAbsent(containerHelixTag, new AtomicInteger(0));
```

Comment:
Not sure if this is still needed, as it should already been added in line 483 inside requestTargetNumberOfContainers. But no harm to add though

Reply:
Oh actually. I think this code is needed because we have code for requesting initial containers. And that does not go through the line 483. I think that is separate